pkg/tls/renewer.go

```diff
@@ -108,7 +108,22 @@ func (c *certRenewer) RenewCA(ctx context.Context) error {
 		logger.Error(err, "failed to write CA")
 		return err
 	}
+
 	logger.Info("CA was renewed")
+
+	_, _, tlsCert, err := c.decodeTLSSecret(ctx)
+	if err != nil && !apierrors.IsNotFound(err) {
+		logger.Error(err, "failed to read TLS")
+		return err
+	}
+	if err = c.verifyCertChain(ctx, caCert, tlsCert); err != nil {
+		logger.Info("mismatched certs chain, renewing", "CA certificate", GenerateRootCASecretName(), "TLS certificate", GenerateTLSPairSecretName())
+		if err := c.RenewTLS(ctx); err != nil {
+			logger.Error(err, "failed to renew TLS certificate", "name", GenerateTLSPairSecretName())
+			return err
+		}
+	}
+
```

Comment:
Do we need this ?

Reply:
Yes, in case of the CA is removed, we want to validate the TLS cert and make sure they are in sync.